src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -202,6 +210,9 @@ def settings_key(self) -> DetectorType:
     def visit_span(self, span: Span) -> None:
         raise NotImplementedError
 
+    def visited_all_spans(self) -> None:
+        pass
```

Comment:
This is being used as more of an exit condition to the for loop to cleanup, should we switch this to some 'clean up detector' hook? Didn't add `visit_all` originally since it's pretty easy to do `n^2` in nested functions if you aren't paying attention, not that it matters now that we are measuring the performance of these.

Reply:
Sure. Sorry is your suggestion around the name, or is it structural? Basically yeah I want to be able to signal to a detector that it has seen the last span and should clean up or emit any last performance issues.

Reply:
Neither really, if you think we'll need all structurally then that works for me too, but we are just using it to clean up the final issue w/o a span to fire, I'd prob rename, up to you.

Reply:
I meant it in past tense to signal that we were done (telling the caller "all spans have been visited") rather than a signal to operate an all spans at once - so that's a naming problem yeah. Do you have any suggestions? `cleanup`? `on_complete`? 🤷

Reply:
Something `on_` is always pretty clear, `on_complete` works for me